instruments/tektronix/tektds5xx.py

```diff
@@ -432,7 +431,7 @@ def data_source(self):
     @data_source.setter
     def data_source(self, newval):
         if isinstance(newval, _TekTDS5xxDataSource):
-            newval = TekTDS5xx.Source[newval.name]
+            newval = TekTDS5xx.Source(newval.name)
```

Comment:
name versus value error - this can't have worked before...

Reply:
We use to use a different enum impl so there might be differences due to that

Reply:
That makes sense, thanks for the explanation. I've seen this issue before and was a bit puzzled by it.